### DISCUSSION TITLE
Index update problem

### DISCUSSION THREAD

**Discussion by amirxhero** [Jeroen-G/Explorer#166] [Q&A]
**After creating each record, it must be updated to include the changes in Elasticsearch**

And for nested keys, it must be updated twice so that the elastic search data is synced with the database data.

```php

    public function mappableAs(): array
    {
        return [
            'id' => 'keyword',
            'title_fa' => [
                'type' => 'text',
            ],
            'variants' => 'nested',
        ];
    }
```

For example, if we change the price value of each variant, it will not be updated for the first time in Elasticsearch

**Comment by Jeroen-G** [comment]
Maybe because Scout is not triggered for the parent model if one of the variants has changed.
